crates/store/src/state.rs

```diff
@@ -703,7 +746,12 @@ impl State {
 
     /// Returns the latest block number.
     pub async fn latest_block_num(&self) -> BlockNumber {
-        (self.inner.read().await.chain_mmr.forest() + 1) as BlockNumber
+        Self::compute_latest_block_number(self.inner.read().await.deref())
+    }
+
+    /// Returns the latest block number for the given inner state.
+    fn compute_latest_block_number(inner_state: &InnerState) -> BlockNumber {
+        (inner_state.chain_mmr.forest() + 1).try_into().expect("block number overflow")
     }
```

Comment:
Should this be a method on the `InnerState` instead? (e.g., something like `InnerState::latest_block_num()`).